privilege/privileges/cache.go

```diff
@@ -580,29 +607,52 @@ func (p *MySQLPrivilege) matchColumns(user, host, db, table, column string) *col
 }
 
 // RequestVerification checks whether the user have sufficient privileges to do the operation.
-func (p *MySQLPrivilege) RequestVerification(user, host, db, table, column string, priv mysql.PrivilegeType) bool {
-	record1 := p.matchUser(user, host)
-	if record1 != nil && record1.Privileges&priv > 0 {
+func (p *MySQLPrivilege) RequestVerification(activeRoles []*auth.RoleIdentity, user, host, db, table, column string, priv mysql.PrivilegeType) bool {
```

Comment:
The more things a function does, the less easier to combinate the function.
Could we keep the original interface, and wrap it to provide the new function? 

```
func RequestVerification(p *MySQLPrivilege, activeRoles []*auth.RoleIdentity, ...) {
    for _, role := range activeRoles {
        p.RequestVerification(p, role.user, role.host, ...)
    }
}
```